### PR TITLE
feat: old content warning

### LINKED ISSUE #250: Mark a post as outdated automatically after a period.
**Is your feature request related to a problem? Please describe.**
Not to a theme-related problem but related to a problem in my field of expertise. I use this theme for a development-related blog. Now, as you might know, in our dev world things can go stale really fast if we're talking about actual code posts. Frameworks change, best practices change, and so forth.
This means any post older than x months might be outdated and not be correct anymore.

**Describe the solution you'd like**
So I would like to add an option in the Front Matter which you can toggle to mark the content as a type that can go stale. If this option is active then an older post, starting from a year f.e., would have a small note at the top that the content might be scale.

**Describe alternatives you've considered**
Well, you could delete the content but then you have broken links and it does not showcase your expertise anymore.
You could also manually add the note.
Both, however, require manual labor which we would like to avoid.

**Additional context**
Another dev I know added it to their site: https://brunty.me/post/display-message-on-old-hugo-entries/
If you want I'm more than willing to add this myself, but I wanted to check in first what your thoughts are on this subject 😄

### PR DESCRIPTION
**Pull request by ericswpark**
WIP, needs styling on the alert div. Basic functionality works

Closes #250

### CODE REVIEW

**Comment by ericswpark** [comment]
Forgot to mention but thanks to https://brunty.me/post/display-message-on-old-hugo-entries/ for the original code.

@lxndrblz @SchabrechtsK please let me know if this implementation is okay and please help me with styling. I'm not sure what will look good. Right now the text is inserted right above the content, without any visual distinction which might look odd.

**Comment by lxndrblz** [comment]
@ericswpark Thanks! I'll take care of the CSS styling.

**Comment by ericswpark** [comment]
I changed the implementation so it now displays the age of the post rather than the duration cutoff, which looks better IMHO at conveying the age of the post. Also moved the string to i18n.

Should we consider allowing users to customize the outdated content message shown? Or should we just display the generalized outdated statement?

**Comment by lxndrblz** [comment]
@ericswpark I would leave it a default message and not customizable. 

Personally, I would change the wording to say something like "over xxx days" old, because the problem of this static approach is that if a user does not rebuild their sites daily, the information would be incorrect.

**Comment by lxndrblz** [comment]
@ericswpark Just pushed some basic CSS styling:

image

What do you think?

**Comment by ericswpark** [comment]
@lxndrblz 

> Personally, I would change the wording to include the value set in the old content duration and emphasis the "over", because the problem of this static approach is that if a user does not rebuild their sites daily, the information would be incorrect.

I haven't even considered that! I'll change it back to the original wording.

> ...
> image
> 
> What do you think?

Looks good. Maybe we can make it yellow-colored to better distinguish it as a warning on the page.

**Comment by ericswpark** [comment]
@lxndrblz here's the alert with the new changes:

image

~~The "!" indicator wasn't centered vertically in the circle and I couldn't find a way to center it with padding so I just changed the height to center the indicator~~ I totally meant to make it sideways-oval shaped; it looks better right?

Also added fix for redefined post section name.

File: assets/css/style.css [hunk 1]
```diff
@@ -1041,6 +1042,7 @@ a.btn {
   padding: 0.5rem;
   height: 1.5rem;
   width: 2.5rem;
+  height: 2.5rem;
```
Comment (ericswpark) [inline assets/css/style.css:1045]: @lxndrblz was this added by mistake? Now there are two height attributes

Comment (lxndrblz) [reply assets/css/style.css:1045]: No, that was by mistake. I forgot to remove the lower value.

**Comment by lxndrblz** [comment]
We should also add the remaining translations before this goes live.

File: assets/css/style.css (hunk 1 above)
Comment (ericswpark) [reply assets/css/style.css:1045]: Hmm, then should we find a way to vertically center the "!" mark in the circle?

Comment (lxndrblz) [reply assets/css/style.css:1045]: It should be resolved by setting the height identical to the width. Or am I am missing something?

Comment (ericswpark) [reply assets/css/style.css:1045]: Actually never mind, I thought it wasn't centered vertically in the screenshot but it is. I wonder if I only saw it on my machine but I remember the "!" symbol was not properly vertically centered in the circle.

**Comment by ericswpark** [comment]
@lxndrblz wouldn't it be better to make it a pinned issue? Translators can't tack on translation commits to this PR unless they are a maintainer of the repository and getting all the translations in will take a lot of time unless we can contact all the previous translators. Since the feature is disabled by default I think translations will trickle in as people enable it and see that there is no translated string.

**Review by lxndrblz — APPROVED** [review]
@ericswpark LGTM! Thanks for all your help.

**Comment by SchabrechtsK** [comment]
Very nice! Thanks for the hard work @ericswpark !   
Personal circumstances prevented me from working at all. Great to see it in here 😄

**Pull request merged**